### DISCUSSION TITLE
"Oops! Could not connect to this Shlink server" when adding a server at https://app.shlink.io

### DISCUSSION THREAD

**Discussion by raymond0517** [shlinkio/discussions#118] [Help wanted] [answered]
This interface has 3 input fields
1) name
2) URL
3) API key

Currently I have got shlink server installed at my server.
I tested with CLI commands like

sudo -u www-data php7.4 /var/www/html/shlink/bin/cli short-url:list

sudo -u www-data php7.4 /var/www/html/shlink/bin/cli short-url:generate

they are working fine.

I tried typing in Short URL generated at the browser,
my server will be able to redirect the short URL to actual URL.

Not sure how to setup the shlink Client at https://app.shlink.io/

I tried to enter the 3 input fields value as follow

name : shlink
URL : my base domain name URL where all my short URL is based on. For example, if short URL generated is http ://my.domain/asbef, then I put in http://my.domain as the value of this URL field
API Key: I key in this value first by running sudo -u www-data php7.4 /var/www/html/shlink/bin/cli api-key:generate, copy and paste the API key to this input field

But I keep getting an error
"Oops! Could not connect to this Shlink server.
Make sure you have internet connection, and the server is properly configured and on-line."

How do proper setup of shlink client so that it can interact with shlink server endpoint?

Thank you

**Comment by acelaya** [comment, marked as answer]
There's a couple of things you can check.

First of all, Shlink has not supported PHP 7.4 for quite some time. The minimum supported version in current stable Shlink is PHP 8.1, but next version will only support 8.2, so I would recommend updating to that version.

Then, you won't be able to connect to a Shlink server that is not served with HTTPS while using https://app.shlink.io, which is served with HTTPS, because of [content restrictions in the browser](https://shlink.io/documentation/troubleshooting/#i-see-mixed-content-errors-in-the-browsers-console) (if the http was just an example, then ignore this point).

And finally, please check the server response from the browser network tab to see the actual error, and the logs generated by Shlink. Those two should throw some light on the root cause.

**Reply by raymond0517** [reply]
Hi acelaya, thank you for the reply.

at the console tab under Inspect:
index-cbf6e3a0.js:892 Mixed Content: The page at 'https://app.shlink.io/server/b12bce6a-f3c0-4b3e-add9-4bb7cb5f9387' was loaded over HTTPS, but requested an insecure resource 'http://....'. This request has been blocked; the content must be served over HTTPS.

So, if I make my subdomain a secure website, then I would be able to use https://app.shlink.io to add a server?

Just to confirm the 3 input fields
1) name
2) URL : my subdomain name
3) API key : API key generated using shlink server
have I entered the correct value as mentioned in my first post?

**Reply by acelaya** [reply]
> So, if I make my subdomain a secure website, then I would be able to use https://app.shlink.io to add a server?

Correct, but you still need to update your PHP version. Otherwise, you'll run into other problems

**Reply by raymond0517** [reply]
TQ very much acalaya
